Subsystems/cmdUtil/cmdUtil.c

```diff
@@ -1,545 +1,913 @@
+                            tempuint8);
+                    exit(EXIT_FAILURE);
+                }
+                CopyData(cmd.Packet, &startbyte, (char *)&tempuint8, sizeof(tempuint8));
                 break;
 
-            case 'I':
-                printf("Pkt ID: %s\n", (char *)optarg);
-                tempShort = (short)strtol(optarg, 0, 0);
+            case 'i': /* int16b */
+                forcebigendian = true;
 
-                /*
-                ** The packet ID is supposed to be in Big Endian order
-                ** If this is running on a little endian host, swap it
-                */
-                if (hostByteOrder == LITTLE_ENDIAN)
+            case 'h': /* int16 */
+                templl    = strtoll(optarg, &tail, 0);
+                tempint16 = templl;
+                if (tempint16 != templl)
                 {
-                    byteSwap((char *)&tempShort, sizeof(tempShort));
+                    fprintf(stderr, "ERROR %s:%u - Parameter not int16 %lld -> %d\n", __func__, __LINE__, templl,
+                            tempint16);
+                    exit(EXIT_FAILURE);
                 }
-                memcpy(&CommandData.PacketHdr[0], &tempShort, sizeof(tempShort));
-                CommandData.GotPktId = 1;
-                break;
 
-            case 'v':
-                printf("Verbose messages on.\n");
-                CommandData.Verbose = 1;
+                /* Endian conversion */
+                if (cmd.BigEndian || forcebigendian)
+                    tempint16 = htobe16(tempint16);
+                else
+                    tempint16 = htole16(tempint16);
+
+                CopyData(cmd.Packet, &startbyte, (char *)&tempint16, sizeof(tempint16));
                 break;
 
-            case 'C':
-                tempShort = strtol(optarg, NULL, 0);
-                if (tempShort > 0x7F)
+            case 'w': /* uint16b */
+                forcebigendian = true;
+
+            case 'n': /* uint16 */
+                tempull    = strtoull(optarg, &tail, 0);
+                tempuint16 = tempull;
+                if (tempuint16 != tempull)
                 {
-                    fprintf(stderr, "Command Code Argument: '%s' rejected. Number is too large.\n", optarg);
-                    break;
+                    fprintf(stderr, "ERROR %s:%u - Parameter not uint16 %llu -> %u\n", __func__, __LINE__, tempull,
+                            tempuint16);
+                    exit(EXIT_FAILURE);
                 }
 
-                CommandData.PacketHdr[6] = tempShort;
-                CommandData.GotCmdCode   = 1;
+                /* Endian conversion */
+                if (cmd.BigEndian || forcebigendian)
+                    tempuint16 = htobe16(tempuint16);
+                else
+                    tempuint16 = htole16(tempuint16);
+
+                CopyData(cmd.Packet, &startbyte, (char *)&tempuint16, sizeof(tempuint16));
```

Comment:
Wouldn't it be easier/simpler to pass an extra parameter to `CopyOut` that indicated the byte order?  That is, either "passthrough" (for native values), "big" to write bytes in network order, or "little" to write in inverse network order.  This could be implemented in an endian-neutral code to avoid depending on these non-posix functions.

Reply:
Good suggestion!  I'll write an enhancement issue.

Reply:
See #95